pkg/signup/service/signup_service.go

```diff
@@ -612,58 +614,59 @@ func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, username, phoneNumberOrHas
-	}
-
-	requirements := []labels.Requirement{*sbSelector}
-
-	sbs, err := provider.ListSpaceBindings(requirements...)
-	if err != nil {
-		log.Errorf(nil, err, "unable to list spacebindings for signup %s", signup.Name)
-		return ""
-	}
-
-	// iterate through the SpaceBindings to find the Spaces that the user has access to, then look for the default namespace with a preference for a Space created by the user
-	var defaultNamespace string
-	var createdSpaceFound bool
-	for _, sb := range sbs {
-		spaceName := sb.Labels[toolchainv1alpha1.SpaceBindingSpaceLabelKey]
-		if spaceName == "" { // space may not be initialized
-			// log error and continue so that the api behaves in a best effort manner
-			log.Errorf(nil, fmt.Errorf("spacebinding has no '%s' label", toolchainv1alpha1.SpaceBindingSpaceLabelKey), "unable to get space '%s'", spaceName)
-			continue
-		}
-		space, err := provider.GetSpace(spaceName)
+// GetDefaultUserTarget retrieves the target cluster and the default namespace from the Space a user has access to.
+// If no spaceName is provided (assuming that this is the home space the target information should be taken from)
+// then the logic lists all Spaces user has access to and picks the first one.
+// returned values are:
+//  1. name of the member cluster the Space is provisioned to
+//  2. the name of the default namespace
+//
+// If the user doesn't have access to any Space, then empty strings are returned
+func GetDefaultUserTarget(provider ResourceProvider, spaceName, murName string) (string, string) {
+	if spaceName == "" {
+		sbSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey, selection.Equals, []string{murName})
 		if err != nil {
-			// log error and continue so that the api behaves in a best effort manner
-			// ie. if a space isn't listed something went wrong but we still want to return the other spaces if possible
-			log.Errorf(nil, err, "unable to get space '%s'", spaceName)
-			continue
+			log.Errorf(nil, err, "unable to create spacebindings selector for MUR %s", murName)
+			return "", ""
 		}
 
-		createdSpaceFound = space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey] == signup.Name
+		requirements := []labels.Requirement{*sbSelector}
 
-		for _, ns := range space.Status.ProvisionedNamespaces {
-			if ns.Type == toolchainv1alpha1.NamespaceTypeDefault {
-				// use this namespace if it is the first one found or if the space was created by the user
-				if defaultNamespace == "" || createdSpaceFound {
-					defaultNamespace = ns.Name
-				}
-				break
+		sbs, err := provider.ListSpaceBindings(requirements...)
+		if err != nil {
+			log.Errorf(nil, err, "unable to list spacebindings for MUR %s", murName)
+			return "", ""
+		}
+		if len(sbs) > 0 {
+			spaceNames := make([]string, len(sbs))
+			for i, sb := range sbs {
+				spaceNames[i] = sb.Spec.Space
 			}
+			sort.Strings(spaceNames)
+			spaceName = spaceNames[0]
 		}
+	}
+	if spaceName == "" {
+		return "", ""
+	}
```

Comment:
alternatively you could return if `sbs` are empty and avoid double checking the `spaceName`
```suggestion
		if len(sbs) == 0 {
			return "", ""
		}
		spaceNames := make([]string, len(sbs))
		for i, sb := range sbs {
			spaceNames[i] = sb.Spec.Space
		}
		sort.Strings(spaceNames)
		spaceName = spaceNames[0]
	}
```

Reply:
yeah, no idea why I didn't realize it before: [ca51a4d](https://github.com/codeready-toolchain/registration-service/pull/441/commits/ca51a4d5ef26fd19c1c3e90d5b47540028aec397)
Thanks